shared/src/commonMain/kotlin/io/customer/shared/serializer/CustomAttributeSerializer.kt

```diff
@@ -0,0 +1,20 @@
+package io.customer.shared.serializer
+
+import io.customer.shared.common.CustomAttributes
+
+/**
+ * Serializer interface to support custom data models in [CustomAttributes] that are not available
+ * and/or cannot be parsed by the SDK.
+ */
+interface CustomAttributeSerializer {
```

Comment:
While discussing KMM in a call, Jatin suggested to have custom serializer where client apps can send their objects by providing implementation for parsing them to/from json. This interface will help us achieve this.

Reply:
I don't understand the need for this as `JsonAdapter` should be doing this, right?

Reply:
`JsonAdapter` is actually [registering this](https://github.com/customerio/mobile-shared/pull/4/files#diff-53a8e782f9e34288b0ea596f001a7fc0b2c44bdce0827801e7a9b565961fe6b5R47). But this lets customer pass custom objects to attributes. I have shared an [example for its usage](https://github.com/customerio/mobile-shared/pull/4#discussion_r1041071176) that might help understanding its need.